src/familiar.ts

```diff
@@ -281,3 +384,182 @@ export function timeToMeatify(): boolean {
 
   return false;
 }
+
+type MarginalFamiliar = GeneralFamiliar & {
+  marginalValue: number;
+};
+
+const outfitSlots = $slots`hat, back, shirt, weapon, off-hand, pants, acc1, acc2, acc3, familiar`;
+const cachedOutfits = new Map<number, { weight: number; meat: number; bonus: number }>();
+
+function cacheOutfit(
+  mult: number,
+  familiar: Familiar
+): { weight: number; meat: number; bonus: number } {
+  // Ignore free fight equips in this meat outfit valuation
+  const noFreeFightsReq = new Requirement([], {
+    preventEquip: $items`Kramco Sausage-o-Matic™, cursed magnifying glass, protonic accelerator pack, "I Voted!" sticker`,
+  });
+
+  useFamiliar(familiar);
+  meatOutfit(false, noFreeFightsReq, false);
+
+  const outfit = outfitSlots.map((slot) => equippedItem(slot));
+  const bonuses = bonusGear("barf");
+  const values = {
+    weight: sum(outfit, (eq: Item) => getModifier("Familiar Weight", eq)),
+    meat: sum(outfit, (eq: Item) => getModifier("Meat Drop", eq)),
+    bonus: sum(outfit, (eq: Item) => bonuses.get(eq) ?? 0),
+  };
+  cachedOutfits.set(mult, values);
+  return values;
+}
+
+export function setMarginalFamiliar(loc: Location): void {
```

Comment:
This function desperately needs splitting up. I'd consider [reading about the SLA principle](https://medium.com/javarevisited/slap-that-ugly-code-6ec276d3a4bc) and thinking about how many layers of abstraction this function has currently.